chain/ethereum/src/chain.rs

```diff
@@ -840,13 +969,25 @@ impl TriggersAdapterTrait<Chain> for TriggersAdapter {
     }
 
     async fn is_on_main_chain(&self, ptr: BlockPtr) -> Result<bool, Error> {
-        self.chain_client
-            .rpc()?
-            .cheapest()
-            .await
-            .ok_or(anyhow!("unable to get adapter for is_on_main_chain"))?
-            .is_on_main_chain(&self.logger, ptr.clone())
-            .await
+        match &*self.chain_client {
```

Comment:
wondering if this should be on the ChainClient, since it's implemented already for rpc adapters, perhaps it would make sense to get the block from the adapter (whichever) and then have the logic on the layer above that makes the decision on whether it is on the main chain.

Reply:
Would be hard to move since, the `adapter.cheapest()` method is only implemented for `EthereumNetworkAdapters` its not on the trait. Its not part of the `Blockchain::Client` which is what is available in the `ChainClient` implmentation.